src/core/reducers/addons.js

```diff
@@ -37,69 +58,151 @@ export function fetchAddon({ errorHandler, slug }: FetchAddonParams): FetchAddon
+    status: apiAddon.status,
+    summary: apiAddon.summary,
+    support_email: apiAddon.support_email,
+    support_url: apiAddon.support_url,
+    tags: apiAddon.tags,
+    type: apiAddon.type,
+    url: apiAddon.url,
+    weekly_downloads: apiAddon.weekly_downloads,
+
+    // These are custom properties not in the API response.
+
+    // TODO: remove this if possible. I think it was added by mistake
+    // but there are some things relying on it :/
+    iconUrl: apiAddon.icon_url,
+
+    isRestartRequired: false,
+  };
+
+  if (addon.type === ADDON_TYPE_THEME && apiAddon.theme_data) {
+    // This merges theme_data into the addon.
+    // TODO: Let's stop doing that because it's confusing. Lots of
+    // deep button / install code will need to be fixed.
+    addon = {
+      ...addon,
+      ...removeUndefinedProps({
+        accentcolor: apiAddon.theme_data.accentcolor,
+        author: apiAddon.theme_data.author,
+        category: apiAddon.theme_data.category,
+
+        // TODO: Set this back to apiAddon.theme_data.description after
+        // https://github.com/mozilla/addons-frontend/issues/1416
+        // is fixed.
+        // theme_data will contain `description: 'None'` when the
+        // description
+        // is actually `null` and we don't want to set that on the addon
+        // itself so we reset it in case it's been overwritten.
+        //
+        // See also https://github.com/mozilla/addons-server/issues/5650.
+        description: apiAddon.description,
+
+        detailURL: apiAddon.theme_data.detailURL,
+        footer: apiAddon.theme_data.footer,
+        footerURL: apiAddon.theme_data.footerURL,
+        header: apiAddon.theme_data.header,
+        headerURL: apiAddon.theme_data.headerURL,
+        iconURL: apiAddon.theme_data.iconURL,
+        id: apiAddon.theme_data.id,
+        name: apiAddon.theme_data.name,
+        previewURL: apiAddon.theme_data.previewURL,
+        textcolor: apiAddon.theme_data.textcolor,
+        updateURL: apiAddon.theme_data.updateURL,
+        version: apiAddon.theme_data.version,
+      }),
     };
   }
-  return apiAddon;
+
+  if (apiAddon.current_version && apiAddon.current_version.files.length > 0) {
+    // TODO: support specific platform files.
+    // See https://github.com/mozilla/addons-frontend/issues/2998
+    addon.installURL = apiAddon.current_version.files[0].url || '';
+    addon.isRestartRequired = apiAddon.current_version.files.some(
+      (file) => !!file.is_restart_required
+    );
+  }
+
+  // Remove undefined properties entirely. This is for some legacy code
+  // in Discopane that relies on spreads to combine a Discopane result
+  // (which has a header and description) with a minimal add-on object.
+  // For example, the minimal add-on object does not have a description
+  // property so the spread should not override `description`.
+  addon = removeUndefinedProps(addon);
+
+  return addon;
 }
 
-export default function addon(
+const initialState = {};
```

Comment:
Flow type?

Reply:
There is a type for it -- you can see it assigned in the function signature: https://github.com/mozilla/addons-frontend/pull/3073/files/39fb66595b5b8033cd43288e61a19f35d9f32dbf#diff-21aa4f8ea4fba6406233649d0ce65beaR193